vhdbuilder/packer/storage-scan.sh

```diff
@@ -0,0 +1,28 @@
+#!/usr/bin/env bash
+set -euxo pipefail
+
+STORAGE_REPORT_PATH=/opt/azure/containers/storage-report.txt
+
+mkdir -p "$(dirname "${STORAGE_REPORT_PATH}")"
+
+echo "----" >> $STORAGE_REPORT_PATH
+{ df -h >> "$STORAGE_REPORT_PATH" ; } 2>&1
+echo "----" >> $STORAGE_REPORT_PATH
+
+CUR_DIR=$(pwd)
+
+mkdir -p /mnt/sdb1
+mount /dev/sdb1 /mnt/sdb1
```

Comment:
I know this is draft but I just wanted to mention this since I've dealt with it before: you can't rely on a particular device having a particular label within the context of Azure VMs - you need to determine which device is actually the OS device at runtime: https://learn.microsoft.com/en-us/troubleshoot/azure/virtual-machines/linux/troubleshoot-device-names-problems#discover-filesystem-uuids-by-using-blkid